### DISCUSSION TITLE
Snapshots of SwiftUI list weird rendering

### DISCUSSION THREAD

**Discussion by fespinoza** [pointfreeco/swift-snapshot-testing#781] [Q&A]
Hi,

I haven't found a topic for this before.

Does any of you know **why the snapshots of SwiftUI list look like this?** can this be prevented?

<img src="https://github.com/pointfreeco/swift-snapshot-testing/assets/167989/be48fb9a-1120-43cd-b14c-b015220926f8" width="350" />

based on a test like

```swift

func testOutlineView() throws {
        let rootView = NavigationView {
            OutlineGroupExperimentsView()
        }

        let bundleURL = try XCTUnwrap(Bundle(for: type(of: self)).resourceURL)
        assertSnapshot(view: rootView, testBundleResourceURL: bundleURL)
    }
```

the `assertSnapshot` method comes from https://gist.github.com/jaanus/7e14b31f7f445435aadac09d24397da8 for running in Xcode Cloud.

the UI implementation can be found in https://github.com/fespinoza/Prototype-House/blob/main/Prototype%20House/Prototypes/OutlineGroup%20Experiments/OutlineGroupExperimentsView.swift

Snapshot taken with the latest version of the library on Xcode 15 / iOS 17 / iPhone 15 Pro.

---

Also, when it comes to list, when taking the snapshots of list on an iPhone device, but with the `.image(on: .iPad_11inch)` settings, sometimes™️, I see the snapshots not cover the full width of the screen, like:

<img src="https://github.com/pointfreeco/swift-snapshot-testing/assets/167989/400e2507-75b3-4c83-8fbd-ced74eb58e25" width=600 />

**Comment by DavidBrunow** [comment]
To get more accurate rendering you need to run your snapshot tests in a host application and set `drawHierarchyInKeyWindow` to `true`.
